packages/chain-adapters/src/solana/SolanaChainAdapter.ts

```diff
@@ -433,40 +462,49 @@ export class ChainAdapter implements IChainAdapter<KnownChainIds.SolanaMainnet>
 
     // Set compute unit limit to the maximum compute units for the purposes of estimating the compute unit cost of a transaction,
     // ensuring the transaction does not exceed the maximum compute units alotted for a single transaction.
-    instructions.push(ComputeBudgetProgram.setComputeUnitLimit({ units: MAX_COMPUTE_UNITS }))
+    estimationInstructions.push(
+      ComputeBudgetProgram.setComputeUnitLimit({ units: MAX_COMPUTE_UNITS }),
+    )
 
     // placeholder compute unit price instruction for the purposes of estimating the compute unit cost of a transaction
-    instructions.push(ComputeBudgetProgram.setComputeUnitPrice({ microLamports: 0 }))
+    estimationInstructions.push(ComputeBudgetProgram.setComputeUnitPrice({ microLamports: 0 }))
 
     const message = new TransactionMessage({
       payerKey: new PublicKey(input.chainSpecific.from),
-      instructions,
+      instructions: estimationInstructions,
       // static block hash as fee estimation replaces the block hash with latest to save us a client side call
       recentBlockhash: '4sGjMW1sUnHzSxGspuhpqLDx6wiyjNtZAMdL4VZHirAn',
-    }).compileToV0Message()
+    }).compileToV0Message(addressLookupTableAccounts)
 
     const transaction = new VersionedTransaction(message)
 
     return Buffer.from(transaction.serialize()).toString('base64')
   }
 
-  private async buildTokenTransferInstructions({
+  public async createAssociatedTokenAccountInstruction({
     from,
     to,
     tokenId,
-    value,
   }: {
     from: string
     to: string
     tokenId: string
-    value: string
-  }): Promise<TransactionInstruction[]> {
-    const instructions: TransactionInstruction[] = []
+  }): Promise<{
+    instruction?: TransactionInstruction
+    destinationTokenAccount: PublicKey
+  }> {
+    const accountInfo = await this.connection.getAccountInfo(new PublicKey(tokenId))
+
+    const TOKEN_PROGRAM =
+      accountInfo?.owner.toString() === TOKEN_2022_PROGRAM_ID.toString()
+        ? TOKEN_2022_PROGRAM_ID
+        : TOKEN_PROGRAM_ID
```

Comment:
This is some of the meaty logic:

Sending Token2022 tokens were probably not working because of this, we were always using the SPL token programId (ex: sending catwifhat to someone already owning it or who never owned it)

Now we support both (I don't know any other protocols, I think it's the 2 used globally)

Needed for swapping to a Token2022 token using an external receive address (another edge case)